File: fooddairy-quarkus/src/main/java/org/acme/model/RecipeIngredient.java
package org.acme.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
public class RecipeIngredient extends PanacheEntity {

    @ManyToOne
    @JoinColumn(name = "recipe_id")
    private Recipe recipe;

    @ManyToOne
    @JoinColumn(name = "ingredient_id")
    private Ingredient ingredient;

    private float amount;

    public RecipeIngredient(){

    }

    public RecipeIngredient(Recipe recipe, Ingredient ingredient, float amount){
        this.recipe = recipe;
        this.ingredient = ingredient;
        this.amount = amount;
    }

    public Ingredient getIngredient() {
        return ingredient;
    }

    public Recipe getRecipe() {
        return recipe;
    }

    public void setRecipe(Recipe recipe) {
        this.recipe = recipe;
    }

    public void setIngredient(Ingredient ingredient) {
        this.ingredient = ingredient;
    }

    public void setAmount(float amount) {
        this.amount = amount;
    }

    public float getAmount() {
        return amount;
    }
}

File: FoodDairy/app/src/main/java/com/example/fooddairy/db/RecipeIngredient.kt
package com.example.fooddairy.db

import androidx.room.*

@Entity(primaryKeys = ["recipeId", "ingredientId"],
    /*foreignKeys = [ForeignKey(entity = Recipe::class,
        parentColumns = ["recipe_id"],
        childColumns = ["recipe_fid"]), ForeignKey(entity = Ingredient::class,
        parentColumns = ["ingredient_id"],
        childColumns = ["ingredient_fid"])]*/
)

data class RecipeIngredient(
    //@ColumnInfo(name = "recipe_fid")
    var recipeId: Int,

    //@ColumnInfo(name = "ingredient_fid")
    var ingredientId: Int,

    //@ColumnInfo(name = "amount")
    var amount: Float
)
